src/geometry/euclidean/basic.lean

```diff
@@ -682,4 +705,127 @@ calc dist (r1 • v +ᵥ p1) (r2 • v +ᵥ p2) * dist (r1 • v +ᵥ p1) (r2 
 ... = dist p1 p2 * dist p1 p2 + (r1 - r2) * (r1 - r2) * (∥v∥ * ∥v∥)
   : by { rw [dist_eq_norm_vsub V p1, abs_mul_abs_self, mul_assoc] }
 
+/-- Reflection in an affine subspace, which is expected to be nonempty
+and complete.  The word "reflection" is sometimes understood to mean
+specifically reflection in a codimension-one subspace, and sometimes
+more generally to cover operations such as reflection in a point.  The
+definition here, of reflection in an affine subspace, is a more
+general sense of the word that includes both those common cases.  If
+the subspace is empty or not complete, this uses the identity map
+instead. -/
```

Comment:
This last sentence may be a bit confusing since the fact is hidden under the definition of orthogonal projection, and you probably don't want to prove it. Maybe write something like "due to our definition of orthogonal projection in this case the reflection turns out to be identity" (or a version which sounds like proper English and hopefully also conveys the information that this case if irrelevant).

bors d+